pkg/credentials/sts_web_identity.go

```diff
@@ -121,7 +121,11 @@ func getWebIdentityCredentials(clnt *http.Client, endpoint, roleARN, roleSession
 		v.Set("RoleSessionName", roleSessionName)
 	}
 	v.Set("WebIdentityToken", idToken.Token)
-	v.Set("DurationSeconds", fmt.Sprintf("%d", idToken.Expiry))
+	if idToken.Expiry > 0 {
+		v.Set("DurationSeconds", fmt.Sprintf("%d", idToken.Expiry))
+	} else {
+		v.Set("DurationSeconds", "3600")
```

Comment:
default a default global constant for this @saracen

Reply:
I've updated. The alternative would be to omit DurationSeconds entirely as it is an optional parameter for AWS' implementation. But I don't know whether that's true for Minio's.

Reply:
That is true for MinIO as well @saracen

Reply:
Okay, I'll update shortly to exclude sending the value if it is zero.